src/service/xhr-impl.js

```diff
@@ -147,6 +149,26 @@ export class Xhr {
     return this.fetch(input, setupInit(opt_init, 'text/plain'));
   }
 
+  /**
+   * A subsitute for the standard response.json(), which may optionally strip a prefix before calling JSON.parse().
+   *
+   * @param {!Response} res fetch response to convert to json.
+   * @param {string|undefined} prefix to strip away.
+   * @return {Promise<*>}
```

Comment:
```suggestion
   * @return {!Promise<!JsonObject>}
```

Reply:
I think `Response.json()` returns `Promise<*>` which may include JSON primitives e.g. number, boolean.

Reply:
Yep, `Body.json()` returns a `Promise<any>`. 
<img width="1255" alt="Screen Shot 2019-11-12 at 10 58 34 PM" src="https://user-images.githubusercontent.com/4656974/68732018-ffeecb00-059f-11ea-9aaf-6085b5de2efb.png">
Source: https://fetch.spec.whatwg.org/#dom-body.

You can even see this by running the snippet:
```js
const resp = new Response("5");
const json = await resp.json();
assert.equals(json, 5);
```